source/wifi/wifi_hal.c

```diff
@@ -7200,11 +7199,32 @@ INT wifi_getApAssociatedDeviceTidStatsResult(INT radioIndex,  mac_address_t *cli
 
 INT wifi_startNeighborScan(INT apIndex, wifi_neighborScanMode_t scan_mode, INT dwell_time, UINT chan_num, UINT *chan_list)
 {
-    // TODO Implement me!
+    INT status = RETURN_ERR;
+    UINT index;
+    INT Radio_Index;
+    char cmd[256]={0};
+    WIFI_ENTRY_EXIT_DEBUG("Inside %s:%d\n",__func__, __LINE__);
+    wifi_getApRadioIndex(apIndex,&Radio_Index);	
+    if (scan_mode == WIFI_RADIO_SCAN_MODE_ONCHAN)
+    {   
+        return RETURN_OK;
+    }
+     /* If channels are not specified use default driver params */
+    if (chan_num)
+    {
+        uint32_t    chan_index;
+        for (chan_index = 0; chan_index < chan_num; chan_index++)
+        {   
+     	    snprintf(cmd,sizeof(cmd),  "iw dev %s%d scan -u freq %u passive >> /tmp/%u.txt ", RADIO_PREFIX, Radio_Index, chan_list[chan_index],chan_list[chan_index]);
+	    system(cmd);
+        }
+     }else { 
+         snprintf(cmd, sizeof(cmd),  "iw dev %s%d scan passive >> /tmp/scan_results.txt ", RADIO_PREFIX, Radio_Index );
+         system(cmd);
+     }
     return RETURN_OK;
```

Comment:
You use WIFI_ENTRY_EXIT_DEBUG on entry but there is missing WIFI_ENTRY_EXIT_DEBUG on exit

Reply:
done